src/System.Management.Automation/engine/Modules/SwitchProcessCommand.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Runtime.InteropServices;
+
+using Dbg = System.Management.Automation.Diagnostics;
+
+#if UNIX
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// Implements a cmdlet that allows use of execv API
+    /// </summary>
+    [Experimental(ExperimentalFeature.PSExecFeatureName, ExperimentAction.Show)]
+    [Cmdlet(VerbsCommon.Switch, "Process", HelpUri = "https://replace")]
+    public sealed class SwitchProcessCommand : PSCmdlet
+    {
+        /// <summary>
+        /// Command to execute.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = false, ValueFromPipeline = true, ValueFromRemainingArguments = true)]
+        public string[] WithCommand { get; set; } = Array.Empty<string>();
+
+        /// <summary>
+        /// For each record, execute it, and push the results into the success stream.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            if (WithCommand.Length == 0)
+            {
+                return;
+            }
+
+            // execv requires command to be full path so resolve command to first match
+            var command = this.SessionState.InvokeCommand.GetCommand(WithCommand[0], CommandTypes.Application);
```

Comment:
```suggestion
            CommandInfo command = this.SessionState.InvokeCommand.GetCommand(WithCommand[0], CommandTypes.Application);
```

Reply:
Did you consider `execvp`? I guess we could avoid GetCommand overhead.

Reply:
I did consider that, but the problem is if you did `exec get-command`, you wouldn't get a good error message.  I could still use `execvp` where you would get behavior like continued search if you would get access denied, but thought that might not be desired behavior.

Reply:
I'd expect an error message after `exitCode < 0` below could useful enough if we add the errorcode and full command line.

Reply:
Based on this https://linux.die.net/man/3/execv, it seems that the behavior for `execvp` isn't defined as a standard and just by convention where BSD and Linux decided on different behavior.  So for consistency, I think it's better to have this behavior.  For the error message, I can include the command line where I can see it being useful in scripts.

Reply:
In the docs I see only notes for errors "The behavior of execlp() and execvp() when errors occur while attempting to execute the file is historic practice, but has not traditionally been documented and is not specified by the POSIX standard."

I do not insist on using execvp, but it seems to me there will be less problems if we do exactly what Unix does in this case (including parsing arguments). But we can wait feedbacks.

Reply:
I would probably avoid `execvp` for the reasons that Steve mentions above. The unix shells that I've inspected don't use `execvp`. Which shell have you seen which does?